### PR TITLE
K8s: JSON file storage

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

Adds JSON file storage implementation of `storage.Interface`.

**Why do we need this feature?**

**Who is this feature for?**

@grafana/grafana-app-platform-squad 

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

`config.ini` changes:

```ini
app_mode = development

[feature_toggles]
grafanaAPIServer = true
kubernetesPlaylists = true

[grafana-apiserver]
storage_type = file
```

Objects will be written to disk under the `{data.path}/grafana-apiserver/` directory.

For example:

```
data/grafana-apiserver
├── grafana.kubeconfig
└── playlist.grafana.app
    └── playlists
        └── default
            └── hi.json
```

To test playlists, you can get sample files from: https://github.com/grafana/grafana/pull/75709
```
git checkout origin/playlist-add-resource -- pkg/tests/apis/playlist/testdata/
```
Then run:
```
kubectl create -f ./pkg/tests/apis/playlist/testdata/playlist-generate.yaml
```
a few times, and you will have:
image

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by ryantxu** [comment]
should this be `storage_type = file`?  (and update similar names?) -- regardless where it is saved it will all be JSON 🤣   the peanut butter of k8s

image

**Comment by toddtreece** [comment]
@ryantxu `storage_type = file` sounds good to me. i will update

**Review by ryantxu — APPROVED** [review]
works great now!

File: pkg/apis/playlist/v0alpha1/doc.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // +k8s:deepcopy-gen=package
 // +k8s:openapi-gen=true
-// +groupName=playlist.grafana.io
+// +groupName=playlist.grafana.app
```
Comment (ryantxu) [inline pkg/apis/playlist/v0alpha1/doc.go:3]: unrelated.. but yes 👍🏻

**Pull request merged**